api4/post.go

```diff
@@ -168,6 +169,41 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(c.App.PostListWithProxyAddedToImageURLs(list).ToJson()))
 }
 
+func getPostsForChannelAroundLastUnread(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	channelId := c.Params.ChannelId
+	if !c.App.SessionHasPermissionToChannel(c.Session, channelId, model.PERMISSION_READ_CHANNEL) {
+		c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+		return
+	}
+
+	postList, err := c.App.GetPostsForChannelAroundLastUnread(channelId, c.Session.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	etag := ""
+	if len(postList.Order) == 0 {
+		etag = c.App.GetPostsEtag(channelId)
+
+		if c.HandleEtag(etag, "Get Posts", w, r) {
+			return
+		}
+
+		postList, err = c.App.GetPostsPage(channelId, 0, 60)
```

Comment:
if we don't have it define, can we define the number `60` somewhere as a constant?

Reply:
Now updated, please see changes.